testng-core/src/main/java/org/testng/SuiteRunner.java

```diff
@@ -514,7 +510,12 @@ public String getOutputDirectory() {
 
   @Override
   public Map<String, ISuiteResult> getResults() {
-    return suiteResults;
+    // When users want to access the results, they are NOT expected to alter it's state.
+    // So we don't need to be sending back a synchronized map which adds a small additional cost
+    // to enter and exit the synchronized block.
+    // Also just to ensure that we guard the internals of the suite results we now wrap it
+    // around with an unmodifiable map.
+    return Collections.unmodifiableMap(new LinkedHashMap<>(suiteResults));
```

Comment:
We are returning back a new map, because `suiteResults` is a synchronizedMap which means there's an added cost to entering/exiting the mutex barrier created by the `synchronized` block within the `get` operation of the `SynchronizedMap`.